app/services/common_platform/api/search_prosecution_case.rb

```diff
@@ -11,10 +11,51 @@ def initialize(params)
 
       def call
         check_response_status
-
+        check_for_blank_defendants(prosecution_cases)
         record_prosecution_cases
       end
 
+      def check_for_blank_defendants(prosecution_cases)
+        if !prosecution_cases.nil? && !prosecution_cases.empty?
+          prosecution_cases.each do |prosecution_case|
+            if !prosecution_case["defendantSummary"].nil? && !prosecution_case["defendantSummary"].empty?
+              prosecution_case["defendantSummary"].each do |defendant|
+                # If the defendant is missing these values in their summary
```

Comment:
Rather than commenting what these complex conditions are doing, it is typical in development to pull them out in to their own method with the method name explaining what it is doing. This makes it easier to understand and test. Here the above condition could be a method has_defendant_summary_and_prosecution_case or something?

Reply:
It also makes the comment not necessary.

Reply:
Makes sense. I've split things up into separate methods where appropriate. Let me know if you have any more feedback on this 😄